src/dynatrace/opentelemetry/metrics/export/oneagentmetadataenricher.py

```diff
@@ -19,51 +19,65 @@
 class OneAgentMetadataEnricher:
     __logger = None
 
-    def __init__(self, logger: logging.Logger) -> None:
-        self.__logger = logger
+    def __init__(self) -> None:
+        self.__logger = logging.Logger(self.__class__.__name__)
 
     def add_oneagent_metadata_to_tags(self, tags: Mapping[str, str]):
-        for key, value in self._parse_oneagent_metadata(
-                self.__get_metadata_file_content()).items():
+        metadata_file_content = self._get_metadata_file_content()
+        parsed_metadata = self._parse_oneagent_metadata(metadata_file_content)
+        for key, value in parsed_metadata.items():
             tags[key] = value
 
-    def __get_metadata_file_content(self) -> List[str]:
+    def _get_metadata_file_name(self, indirection_fname: str) -> str:
+        file_name = None
+        if not indirection_fname:
+            return None
+
+        try:
+            with open(indirection_fname, "r") as metadata_indirection_file:
+                file_name = metadata_indirection_file.read()
+
+        except OSError:
+            self.__logger.info("Could not read local OneAgent indirection "
+                               "file. This is normal if no OneAgent is "
+                               "installed.")
+        return file_name
+
+    def _get_metadata_file_content(self) -> List[str]:
         try:
-            metadata_file_name = None
-            with open(
-                    "dt_metadata_e617c525669e072eebe3d0f08212e8f2.properties",
-                    "r") as magic_file:
-                metadata_file_name = magic_file.read()
+            metadata_file_name = self._get_metadata_file_name(
+                "dt_metadata_e617c525669e072eebe3d0f08212e8f2.properties"
+            )
 
             if not metadata_file_name:
-                self.__logger.debug("metadata file name not specified by "
-                                    "OneAgent.")
+                self.__logger.warning("Metadata file not specified in "
+                                      "indirection file!")
                 return []
 
             with open(metadata_file_name, "r") as attributes_file:
                 return attributes_file.readlines()
-
         except OSError:
-            logging.warning(
-                "could not read OneAgent metadata file. This is normal if "
-                "OneAgent is not installed.")
+            self.__logger.info(
+                "Could not read OneAgent metadata file.")
```

Comment:
You could print the file path here so users could check the permissions on the file.